core/runtime/common/runtime_instances_pool.cpp

```diff
@@ -74,63 +74,65 @@ namespace kagome::runtime {
   outcome::result<std::shared_ptr<ModuleInstance>, CompilationError>
   RuntimeInstancesPool::instantiateFromCode(const CodeHash &code_hash,
                                             common::BufferView code_zstd) {
+    std::optional<std::reference_wrapper<InstancePool>> pool_opt{};
+
     std::unique_lock lock{pools_mtx_};
-    auto pool_opt = pools_.get(code_hash);
+    pool_opt = pools_.get(code_hash);
```

Comment:
Revert unnecessary change
```suggestion
    auto pool_opt = pools_.get(code_hash);
```